src/store/Workspaces/devWorkspaces/index.ts

```diff
@@ -217,6 +220,30 @@ export const actionCreators: ActionCreators = {
     }
   },
 
+  restartWorkspace: (workspace: IDevWorkspace): AppThunk<KnownAction, Promise<void>> => async (dispatch, getState): Promise<void> => {
+    const defer: IDeferred<void> = getDefer();
+    try {
+      await devWorkspaceClient.changeWorkspaceStatus(workspace.metadata.namespace, workspace.metadata.name, false);
+      dispatch({ type: 'DELETE_DEVWORKSPACE_LOGS', workspaceId: workspace.status.devworkspaceId });
+      restartCallbacks.set(workspace.status.devworkspaceId, async () => {
+        try {
+          await actionCreators.startWorkspace(workspace)(dispatch, getState, undefined);
+          defer.resolve();
+        } catch (e) {
+          defer.reject(e);
+        }
+      });
+    } catch (e) {
+      const errorMessage = `Failed to restart the workspace with ID: ${workspace.status.devworkspaceId}, reason: ` + getErrorMessage(e);
```

Comment:
is it going to be exposed to users?
if yes - I would replace ID with name, since users do not have workspace ID on UI anywhere
Please check other places where you use workspace id.

Reply:
I have removed this addition part from the target error